File: javascript/167.two-sum-ii-input-array-is-sorted/167.two-sum-ii-input-array-is-sorted.js
/*
 * @lc app=leetcode id=167 lang=javascript
 *
 * [167] Two Sum II - Input Array Is Sorted
 */

// @lc code=start
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    const map = new Map();

    for (let i = 0; i < numbers.length; i++) {
        const substructionResult = target - numbers[i];
        const mapVal = map.get(substructionResult);
        if (mapVal >= 0) {
            return [mapVal + 1, i + 1];
        }
        map.set(numbers[i], i)
    }
    return [0, 0]
};
// @lc code=end

File: python/167.two-sum-ii-input-array-is-sorted/167.two-sum-ii-input-array-is-sorted.py
#
# @lc app=leetcode id=167 lang=python3
#
# [167] Two Sum II - Input Array Is Sorted
#

# @lc code=start
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        r = len(numbers) - 1

        while True:
            sum = numbers[l] + numbers[r]

            if sum == target:
                return [l + 1, r + 1]
            elif sum > target:
                r -= 1
            else:
                l += 1
# @lc code=end
